rotkehlchen/chain/ethereum/decoding/decoder.py

```diff
@@ -522,6 +532,32 @@ def _maybe_enrich_transfers(  # pylint: disable=no-self-use
 
         return None
 
+    def _enrich_protocol_tranfers(  # pylint: disable=no-self-use
+            self,
+            token: Optional[EthereumToken],
+            tx_log: EthereumTxReceiptLog,
+            transaction: EthereumTransaction,
+            event: HistoryBaseEntry,
+            action_items: List[ActionItem],
+    ) -> None:
+        """
+        Decode special transfers made by contract execution for example at the moment
+        of depositing assets or withdrawing.
+        It assumes that the event being decoded has been already filtered and is a
+        transfer.
+        """
+        if (
```

Comment:
this is going to be much better organized if you move the if check inside `enrich_pickle_transfers` and rename the function to `maybe_enrich_pickle_transfers()`

Reply:
The idea is that this function is needed for other protocols too so this would be used for more in theory

Reply:
true. So make the `maybe_` functions return the event or true if it matched and exit the `base_foo()` early.